tests/components/samsungtv/test_media_player.py

```diff
@@ -55,247 +55,316 @@ class UnhandledResponse(Exception):
     """Dummy Exception."""
 
 
-class TestSamsungTv(unittest.TestCase):
-    """Testing Samsungtv component."""
-
-    @MockDependency("samsungctl")
-    @MockDependency("wakeonlan")
-    def setUp(self, samsung_mock, wol_mock):
-        """Set up test environment."""
-        self.hass = tests.common.get_test_home_assistant()
-        self.hass.start()
-        self.hass.block_till_done()
-        self.device = SamsungTVDevice(**WORKING_CONFIG)
-        self.device._exceptions_class = mock.Mock()
-        self.device._exceptions_class.UnhandledResponse = UnhandledResponse
-        self.device._exceptions_class.AccessDenied = AccessDenied
-        self.device._exceptions_class.ConnectionClosed = ConnectionClosed
-
-    def tearDown(self):
-        """Tear down test data."""
-        self.hass.stop()
-
-    @MockDependency("samsungctl")
-    @MockDependency("wakeonlan")
-    def test_setup(self, samsung_mock, wol_mock):
-        """Testing setup of platform."""
-        with mock.patch("homeassistant.components.samsungtv.media_player.socket"):
-            add_entities = mock.Mock()
-            setup_platform(self.hass, WORKING_CONFIG, add_entities)
+@pytest.fixture
+def samsung_mock():
+    """Mock samsungctl."""
+    with MockDependency("samsungctl"):
+        yield
 
-    @MockDependency("samsungctl")
-    @MockDependency("wakeonlan")
-    def test_setup_discovery(self, samsung_mock, wol_mock):
-        """Testing setup of platform with discovery."""
-        with mock.patch("homeassistant.components.samsungtv.media_player.socket"):
-            add_entities = mock.Mock()
-            setup_platform(self.hass, {}, add_entities, discovery_info=DISCOVERY_INFO)
-
-    @MockDependency("samsungctl")
-    @MockDependency("wakeonlan")
-    @mock.patch("homeassistant.components.samsungtv.media_player._LOGGER.warning")
-    def test_setup_none(self, samsung_mock, wol_mock, mocked_warn):
-        """Testing setup of platform with no data."""
-        with mock.patch("homeassistant.components.samsungtv.media_player.socket"):
+
+@pytest.fixture
+def wakeonlan_mock():
+    """Mock wakeonlan."""
+    with MockDependency("wakeonlan"):
+        yield
+
+
+async def test_setup(hass, samsung_mock, wakeonlan_mock):
+    """Test setup of platform."""
+    with mock.patch("homeassistant.components.samsungtv.media_player.socket"):
+        add_entities = mock.Mock()
+        await async_setup_platform(hass, WORKING_CONFIG, add_entities)
```

Comment:
We don't want to use any platform code directly in the tests. We want to use the core to set up the platform via `async_setup_component` and then assert that there is a state corresponding to the entity we expect. Writing tests like this makes them robust for changes in the platform.

The state is the truth of the entity.

```suggestion
        assert await async_setup_component(
            hass, "media_player", {"media_player": {"platform": "samsungtv", **WORKING_CONFIG}}
        )
        state = hass.states.get("media_player.fake")
        assert state
```

Reply:
Thanks for the help. I stumbled through many integration tests and found many different ways of doing this. There should be some kind of state-of-the-art implementation of tests (with the new scaffold maybe?). Am I right that with this the `test_setup_none` function (no config given) will not gonna happen because the `PLATFORM_SCHEMA` will fail before?